Phoenix/Core/Include/Core/Voxels/Blocks.hpp

```diff
@@ -54,21 +56,29 @@ namespace phoenix
 		};
 
 		/// @brief Stores universal block definitions
-		class BlockRegistry
+		class BlockRegistry : public qz::utils::Singleton<BlockRegistry>
 		{
 		private:
-			std::vector<RegisteredBlock> Blocks;
-			int                          i;
+			std::vector<RegisteredBlock> m_Blocks;
+			int                          m_BlockIdCounter;
```

Comment:
ahh, sorry to be a picky arse, but it's `camelCase` - so `m_blocks` and `m_blockIdCounter` (note the lowercase first letter)